week0/hw/Week0-Homework.ipynb

```diff
@@ -0,0 +1,318 @@
+    "\n",
+    "Potassium permanganate (KMnO4) is used to oxidize Fe2+ and Mn2+.\n",
+    "\n",
+    "Calculate the quantity (kg/d *and* lb/hr) of potassium permanganate required, alkalinity consumed as CaCO3, and quantity of sludge produced in total and for each ion (kg/d).\n",
+    "\n",
+    "Use the oxidation reactions for iron and manganese using KMnO4 as shown in Tables 20-8 and 20-10, respectively."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1ea023c7",
+   "metadata": {},
+   "source": [
+    "<img src=\"MWH_Table_20-8.png\">\n",
+    "<img src=\"MWH_Table_20-10.png\">"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2c8bef5f",
+   "metadata": {},
+   "source": [
+    "## Problem statement\n",
+    "\n",
+    "Create a Pyomo model to solve the above problem with the following components:\n",
+    "\n",
+    "- Variables and Constraints\n",
+    "    - Flow rate\n",
+    "    - KMnO4 required for each inlet ion\n",
+    "    - Alkalinity consumed for each inlet ion\n",
+    "    - Sludge produced for each inlet ion\n",
+    "\n",
+    "- Parameters\n",
+    "    - Initial concentration\n",
+    "    - Conversion factors for KMnO4, alkalinity, and sludge from Table 20-8 and 20-10\n",
+    "\n",
+    "- Expressions\n",
+    "    - Total KMnO4 required\n",
+    "    - Total alkalinity consumed\n",
+    "    - Total sludge produced\n",
+    "\n",
+    "The model should print:\n",
+    "- Termination condition\n",
+    "- Total KMnO4 required\n",
+    "- Total alkalinity consumed\n",
+    "- Total sludge produced\n",
+    "- KMnO4 required for each inlet ion\n",
+    "- Alkalinity consumed for each inlet ion\n",
+    "- Sludge produced for each inlet ion\n",
+    "\n",
+    "Components should be indexed to `[\"Fe\", \"Mn\"]` where appropriate. If unit conversions are necessary, use `pyunits` (i.e., no hard-coded unit conversions)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d260be23",
+   "metadata": {},
+   "source": [
+    "## Required data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "35e8244e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Ion index\n",
+    "ions = [\"Fe\", \"Mn\"]\n",
+    "# Initial concentration, g/m3\n",
+    "conc_init = {\"Fe\": 5, \"Mn\": 2}\n",
+    "# Conversion to KMnO4 dose, g/g\n",
+    "kmno4_conv_init = {\"Fe\": 0.94, \"Mn\": 1.92}\n",
+    "# Conversion to alkalinity consumed, g/g\n",
+    "alk_conv_init = {\"Fe\": 1.5, \"Mn\": 1.21}\n",
+    "# Conversion to sludge produced, g/g\n",
+    "sludge_conv_init = {\"Fe\": 2.43, \"Mn\": 2.64}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "596f77b1",
+   "metadata": {},
+   "source": [
+    "## Make necessary imports"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b331695a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import the necessary modules"
```

Comment:
I wouldn't expect them to know the location of the imports at this point, and this step could potentially be very tedious for them w/o offering much educational value. I would just add the imports here for them.